rsocket/statemachine/ConsumerBase.cpp

```diff
@@ -22,7 +22,9 @@ void ConsumerBase::subscribe(
   }
 
   DCHECK(!consumingSubscriber_);
-  consumingSubscriber_ = std::move(subscriber);
+  // keep a refernce to consumingSubscriber_ on the stack in case
+  // its call to onSubscribe causes us to drop our instance reference to it
```

Comment:
How does this, and all follow-up cases help? I'm confused how having the extra reference on the stack is useful, given that this is the last operation we're doing in the function anyway?

Reply:
because the object that we're calling methods on might cause itself to be free'd, and then access uninitialized memory when it references `this`. We can't rely on users adding `auto this_ = get_ref(this)` as a prologue to methods in all of their classes that inherit from Subscription/Subscriber.

Reply:
Yeah you're right... ugh this sucks though.  I guess it's more impetus for `onSubscribe` versus `onSubscribeImpl`.